### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by btang1**
## PR Checklist
- [ ] This PR has been tested on an RDHPCS machine and/or WCOSS2. Please select below:
  - [ ] RDHPCS.
  - [ ] WCOSS2.

- [ ] This PR has been tested with the ufs-srweather-app workflow online-cmaq branch.

- [ ] New or updated input data is required by this PR. 

- [ ] Baselines are expected to change.

### CODE REVIEW

File: src/model/src/ASX_DATA_MOD.F
Comment (ytangnoaa) [inline src/model/src/ASX_DATA_MOD.F:1]: Is this the major change of this PR, removing canopy and adding fengsha?

File: src/model/src/AERO_PHOTDATA.F [hunk 1]
```diff
@@ -2079,25 +2079,25 @@ SUBROUTINE FAST_OPTICS( NR, NI, LAMBDA, DGN, SIGMA_G, BETA_EXT, BETA_SCAT, G )
          Z12   = Z1    * Z1
          Z2    = 4.0   * XNRI2 + 12.0 * XNRMI + 9.0
          XC1   = 8.0   / ( 3.0 * Z12 )
-         A1    = DBLE(24.0  * XRI / Z1) !(Wei Li)
+         A1    = DBLE(24.0  * XRI / Z1)
 
-         A2    = DBLE(44.0   * XRI / 15.0 + 20.0 * XRI / ( 3.0 * Z2 ) +
+         A2    = DBLE(4.0   * XRI / 15.0 + 20.0 * XRI / ( 3.0 * Z2 ) +
```
Comment (bbakernoaa) [inline src/model/src/AERO_PHOTDATA.F:2084]: Double check this... its a big change

Comment (ytangnoaa) [inline src/model/src/AERO_PHOTDATA.F:2084]: Please double check line 2084

File: src/model/src/ASX_DATA_MOD.F [hunk 2]
```diff
@@ -129,24 +129,14 @@ Module ASX_DATA_MOD
          Integer, Allocatable :: LPBL     ( :,: )  ! PBL layer
          Logical, Allocatable :: CONVCT   ( :,: )  ! convection flag
          Real,    Allocatable :: PBL      ( :,: )  ! pbl height (m)
-!         Real,    Allocatable :: NACL_EMIS( :,: )  ! NACL mass emission rate of particles with d <10 um (g/m2/s)
+
+!Real,            Allocatable :: NACL_EMIS( :,: )  ! NACL mass emission rate of particles with d <10 um (g/m2/s)
          Real,    Allocatable :: COSZEN   ( :,: )  ! Cosine of the zenith angle
          Real,    Allocatable :: CFRAC    ( :,: )  ! cloud fraction
 
-!> Inline Canopy Processes (Wei Li)
```
Comment (ytangnoaa) [inline src/model/src/ASX_DATA_MOD.F:136]: We'd better keep the canopy stuff

Comment (drnimbusrain) [reply src/model/src/ASX_DATA_MOD.F:136]: @btang1 Yes, Youhua is correct.  Please keep all inline canopy processes across the AQM codes.

File: src/model/src/ASX_DATA_MOD.F [hunk 3]
```diff
@@ -261,13 +251,9 @@ Module ASX_DATA_MOD
       Real, allocatable, private :: BUFF2D( :,: )   ! 2D temp var
       Real, allocatable, private :: BUFF3D( :,:,: ) ! 3D temp var
 
-! Canopy option control (Wei Li)
```
Comment (drnimbusrain) [inline src/model/src/ASX_DATA_MOD.F:264]: @btang1 Please do not remove any of my canopy mods to the ASX_DATA_MOD.F.  Please revert to include all my canopy changes in updated ASX_DATA_MOD.

File: src/model/src/ASX_DATA_MOD.F [hunk 4]
```diff
@@ -716,33 +702,7 @@ Subroutine INIT_MET ( JDATE, JTIME )
             Grid_Data%BSLP    = 0.0
          End If
 
-!> ccccccccccccccccccccc canopy shade option!ccccccccccccccccccccc (Wei Li)
```
Comment (drnimbusrain) [inline src/model/src/ASX_DATA_MOD.F:719]: @btang1 Please do not remove any of my canopy mods to the ASX_DATA_MOD.F.  Please revert to include all my canopy changes in updated ASX_DATA_MOD.

File: src/model/src/ASX_DATA_MOD.F [hunk 5]
```diff
@@ -941,36 +897,28 @@ Subroutine GET_MET ( JDATE, JTIME, TSTEP )
...
 C-------------------------------- MET_CRO_2D --------------------------------
 C Vegetation and surface vars
+
          call interpolate_var ('LAI', jdate, jtime, Met_Data%LAI)
 
          call interpolate_var ('VEG', jdate, jtime, Met_Data%VEG)
 
          call interpolate_var ('ZRUF', jdate, jtime, Met_Data%Z0)
 
-C Canopy vars (Wei Li)
```
Comment (drnimbusrain) [inline src/model/src/ASX_DATA_MOD.F:952]: Please revert to include all my canopy changes here in updated ASX_DATA_MOD.

**Review by drnimbusrain — CHANGES_REQUESTED** [review]
@btang1 Need to revert and bring back all my canopy changes in the updated ASX_DATA_MOD.F here.

File: src/model/src/AERO_PHOTDATA.F (hunk 1 above)
Comment (btang1) [reply src/model/src/AERO_PHOTDATA.F:2084]: the changed '4' version is correct. double checked the code

Comment (btang1) [reply src/model/src/AERO_PHOTDATA.F:2084]: double checked. the changed '4' version is correct

File: src/model/src/ASX_DATA_MOD.F (hunk 3 above)
Comment (btang1) [reply src/model/src/ASX_DATA_MOD.F:264]: let me double check please

File: src/model/src/ASX_DATA_MOD.F (hunk 5 above)
Comment (btang1) [reply src/model/src/ASX_DATA_MOD.F:952]: I have made updates, move back all canopy inline for ASX_DATA_MOD

File: src/model/src/ASX_DATA_MOD.F (hunk 4 above)
Comment (btang1) [reply src/model/src/ASX_DATA_MOD.F:719]: I have made updates, move back all canopy inline for ASX_DATA_MOD

File: src/model/src/ASX_DATA_MOD.F (hunk 3 above)
Comment (btang1) [reply src/model/src/ASX_DATA_MOD.F:264]: I have made updates, move back all canopy inline for ASX_DATA_MOD

File: src/model/src/ASX_DATA_MOD.F (hunk 2 above)
Comment (btang1) [reply src/model/src/ASX_DATA_MOD.F:136]: I have made updates, move back all canopy inline for ASX_DATA_MOD

File: src/model/src/ASX_DATA_MOD.F
Comment (btang1) [reply src/model/src/ASX_DATA_MOD.F:1]: I have made updates, move back all canopy inline for ASX_DATA_MOD

**Comment by btang1** [comment]
I update ASX_DATA_MOD and Vdiffaxmx

File: src/model/src/o3totcol.f [hunk 6]
```diff
@@ -115,10 +114,10 @@ subroutine o3totcol ( latitude, longitude, jdate, jtime, ozone )
           call m3exit ( pname, jdate, 0, xmsg, xstat1 )
         end if
 
-        ! read nlat, nlon (Wei Li)
+        ! read nlat, nlon
         rewind( tmunit )
-        read( tmunit, *) label, nlat
-        read( tmunit, *) label, nlon
+        read( tmunit, '(5x,i7)') label,nlat
+        read( tmunit, '(5x,i7)') label,nlon
```
Comment (ytangnoaa) [inline src/model/src/o3totcol.f]: This change will cause run-time error, as variable "label" is defined as "character( 8 ) :: label" and '(5x,i7)' only supply one variable's input format instead of two. It's better to reverse back

File: src/model/src/DUST_EMIS.F [hunk 7]
```diff
@@ -837,7 +838,7 @@ subroutine get_dust_emis( jdate, jtime, tstep, rjacm, cellhgt,
  
                 rlay1hgt = rjacm ( c,r ) / cellhgt
  
-                dust_em( c,r ) = dust_alpha * vflux * rlay1hgt *tfa(c,r) * tfb(c,r)
+                dust_em( c,r ) = dust_alpha * vflux * rlay1hgt! *tfa(c,r) * tfb(c,r)
```
Comment (ytangnoaa) [inline src/model/src/DUST_EMIS.F:841]: "*tfa(c,r) * tfb(c,r) " removed. Are you sure?

File: src/model/src/ASX_DATA_MOD.F [hunk 8]
```diff
@@ -261,16 +261,22 @@ Module ASX_DATA_MOD
       Real, allocatable, private :: BUFF2D( :,: )   ! 2D temp var
       Real, allocatable, private :: BUFF3D( :,:,: ) ! 3D temp var
 
-! Canopy option control (Wei Li)
-      CHARACTER( 20 ), SAVE :: CTM_CANOPY_SHADE = 'CTM_CANOPY_SHADE    '! env var for in-line
+! Canopy option control
+      CHARACTER( 20 ), SAVE :: CTM_CANOPY_SHADE = 'CTM_CANOPY_SHADEC'! env var for in-line
```
Comment (ytangnoaa) [inline src/model/src/ASX_DATA_MOD.F]: Why the additional letter "C" is added?
TM_CANOPY_SHADE = 'CTM_CANOPY_SHADE '
vs
TM_CANOPY_SHADE = 'CTM_CANOPY_SHADEC'

File: src/model/src/DUST_EMIS.F (hunk 7 above)
Comment (btang1) [reply src/model/src/DUST_EMIS.F:841]: because tfa and tfb not calculated in the code, so not able to call them here

File: src/model/src/ASX_DATA_MOD.F (hunk 8 above)
Comment (btang1) [reply src/model/src/ASX_DATA_MOD.F]: corrected.

File: src/model/src/o3totcol.f (hunk 6 above)
Comment (btang1) [reply src/model/src/o3totcol.f]: thanks! reversed

**Comment by ytangnoaa** [comment]
Could you fill the following items?

Description
Issue(s) addressed

Have you tested running these changes or just compiled it on Orion?

File: src/model/src/vdiffacmx.F
Comment (ytangnoaa) [inline src/model/src/vdiffacmx.F:1]: This file has many changes brought by CMAQ 5.4+. It is better to highlight the key changes.

Comment (btang1) [reply src/model/src/vdiffacmx.F:1]: sure! let me work on it

Comment (btang1) [reply src/model/src/vdiffacmx.F:1]: compiled vdiffacmx, it compiled without error after I turn off a 'end do' sentence.

**Comment by btang1** [comment]
> 



> Could you fill the following items?
> 
> Description Issue(s) addressed
> 
> Have you tested running these changes or just compiled it on Orion?

all codes I changed have successfully compiled in Orion. I have not run CMAQ online yet. Only compile without error.

**Review by drnimbusrain — APPROVED** [review]
Seems we could approve for now, but there are a lot of changes to vdiffacmx.F, as expected.  However, seems you have added pieces to the old CMAQv5.2.1, rather than remove pieces from CMAQv5.4 version.  I may be OK with either approach, but careful checking may be needed to evaluate if it makes sense.  Also there are numerous places that the bi-directional NH3 dep is removed, and wonder why?  Please elaborate.

File: src/model/src/vdiffacmx.F [hunk 9]
```diff
@@ -275,43 +1072,136 @@ SUBROUTINE VDIFFACMX ( DTSEC, SEDDY, DDEP, ICMP, DDEPJ, DDEPJ_FST, CNGRD )
...
+            END DO
+ 
+#endif
          END DO
 
          DO V = 1, N_SPC_DEPV
             DDEP( V,C,R ) = DDBF( V )
          END DO
- 
-         IF ( ABFLUX ) THEN
```
Comment (drnimbusrain) [inline src/model/src/vdiffacmx.F:310]: @btang1 Why is this removed?

File: src/model/src/vdiffacmx.F [hunk 10]
```diff
@@ -275,43 +1072,136 @@ SUBROUTINE VDIFFACMX ( DTSEC, SEDDY, DDEP, ICMP, DDEPJ, DDEPJ_FST, CNGRD )
...
+                     S_DDBF( V,NP ) = S_DDBF( V,NP ) + THBAR * DD_FAC( V ) * SENS( S,1,NP )
+                  END DO
+#endif
+
 C --------- END of HET HONO RX ----------
 
                ELSE
                   S = DV2DF( V )
                   CONC( S,1 ) = POL( V ) + ( CONC( S,1 ) - POL( V ) ) * EFAC2( V )
-                  DDBF( V ) = DDBF( V ) + THBAR * DD_FAC( V ) * CONC( S,1 )
```
Comment (drnimbusrain) [inline src/model/src/vdiffacmx.F:288]: Why is this removed?

File: src/model/src/vdiffacmx.F [hunk 11]
```diff
@@ -239,32 +721,347 @@ SUBROUTINE VDIFFACMX ( DTSEC, SEDDY, DDEP, ICMP, DDEPJ, DDEPJ_FST, CNGRD )
...
+
+#ifdef sens
+                  DO NP = 1, NPMAX
+                     SENS( S,1,NP ) = S_POL( NP,V ) + ( SENS( S,1,NP ) - S_POL( NP,V ) ) * EFAC1( V )
+                     S_DDBF( V, NP ) = S_DDBF( V, NP ) + THETA * ( DD_FAC( V ) * SENS( S, 1,NP )
+     &                      - DTDENS1 * DD_CONV( V ) * S_PLDV( NP,V,C,R ) )
+                  END DO
+#endif
 
-                  IF ( ABFLUX .AND. V .EQ. NH3_HIT ) THEN
```
Comment (drnimbusrain) [inline src/model/src/vdiffacmx.F:250]: Why removed?

**Review by drnimbusrain — CHANGES_REQUESTED** [review]

**Comment by btang1** [comment]
Hi Youhua,

I have summarized major changes for vdiffacmx. F in the attached file:



 Best wishes,
Beiming

On Aug 29, 2023, at 12:27 PM, Youhua Tang ***@***.***> wrote:



Could you fill the following items?

Description
Issue(s) addressed

Have you tested running these changes or just compiled it on Orion?
